### DISCUSSION TITLE
Can we predict a use case for having artist address in collection and module model?

### DISCUSSION THREAD

**Discussion by reyraa** [Muzikie/blockchain#112] [Ideas]
Before we created the profile module, we added the artist name as a string in the collection and audio module models. Later on we introduced the profile module, which can hold the personal information about the artist. If an account represents a band, the profile can belong to a band. 
My question is, do we still need to store a property in the collection and audio modules to represent the artist? We should taker into account that collections and audios have a `creatorAddress`. Meaning, if you retrieve the information of an audio NFT, you can use the `creatorAddress` to retrieve the profile of the band or artist that created the audio. 
Given the above, I think we no longer need to store the artist name. Although we still need to store co-artists, or featuring artists.In which case we have to make it possible to insert:
- as a string: For co-artists who haven't created an account in Muzikie yet.
- as a Lisk 32 address: For co-artists who already have an account.
We should also make it possible for the creator of the audio to update the co-artists from string to a Lisk 32 address.

**Comment by haghighatcs** [comment]
I think we have to integrate models together just like you mentioned. As we go further well-related models will bring **muzikie** more informative composite data in order to make our queries more accurate and flexible.

**Comment by reyraa** [comment]
In this case:
We need to
- Remove the artistName from Collection
- Remove the artist name from Audio
- Adapt the commands
- Adapt Streamer models
   - Remove the artistName from audios and collections
   - Add reference ID of the profile to the audio and collections
- Read info from Streamer and display the artist info correctly

**Comment by reyraa** [comment]
According to above, if we have
```
coArtistAddresses: ["lskyg9ujmpkbn7ex96ejedhfrkj6avryn5nwgngbp"],
coArtistNames: ["Some Name"]
```
Then users can define an address as a co-artist. We can retrieve the corresponding profile info and show the name and nickName. but alternatively we can show the address if a profile was not created. 
Also if the co-artist does not have a Lisk address, we can address a custom name in the coArtistNames list in string format. Vividly this is less desired and only works as a fallback.
